Plugins: Add feature toggles for long running queries

kevinwcyu opened the pull request:
<!--

Thank you for sending a pull request! Here are some tips:

1. If this is your first time, please read our contribution guide at https://github.com/grafana/grafana/blob/main/CONTRIBUTING.md

2. Ensure you include and run the appropriate tests as part of your Pull Request.

3. In a new feature or configuration option, an update to the documentation is necessary. Everything related to the documentation is under the docs folder in the root of the repository.

4. If the Pull Request is a work in progress, make use of GitHub's "Draft PR" feature and mark it as such.

5. If you can not merge your Pull Request due to a merge conflict, Rebase it. This gets it in sync with the main branch.

6. Name your PR as "<FeatureArea>: Describe your change", e.g. Alerting: Prevent race condition. If it's a fix or feature relevant for the changelog describe the user impact in the title. The PR title is used to auto-generate the changelog for issues marked with the "add to changelog" label.

-->

**What this PR does / why we need it**:
Adds feature toggles for long running queries to be used by the Redshift and Athena plugins.

**Which issue(s) this PR fixes**:

<!--

- Automatically closes linked issue when the Pull Request is merged.

Usage: "Fixes #<issue number>", or "Fixes (paste link of issue)"

-->

Fixes #54093

**Special notes for your reviewer**:

I put the feature area as `Plugins` in the PR title, is that correct?

kevinwcyu committed:
Add feature toggles for long running queries

sunker reviewed (commented):
Feature toggles defined in the grafana.ini are read dynamically, so in theory this is not necessary since they will only be used in the frontend. However, it might be good to define them like this if they're supposed to be used in for example AMG. 

I wonder if it would make sense to replace the `longRunningQueries` suffix from the variable name and replace it with something that describes what the feature adds. For example, `athenaAsyncQueryDataSupport`. WDYT? Also, do you need one for variable for both plugins?

kevinwcyu committed:
update feature flag name

kevinwcyu committed:
make feature toggle frontend only

kevinwcyu commented:
> Feature toggles defined in the grafana.ini are read dynamically, so in theory this is not necessary since they will only be used in the frontend. However, it might be good to define them like this if they're supposed to be used in for example AMG.

Cool, didn't know that. Is this more for documenting it since it's used for AMG, i.e. if we removed these toggles, would it not work in AMG?

I've also updated the toggles to be set to `FrontendOnly: true,` since only the frontend is reading it.

> I wonder if it would make sense to replace the longRunningQueries suffix from the variable name and replace it with something that describes what the feature adds. For example, athenaAsyncQueryDataSupport. WDYT?

I've updated to a different suffix, `AsyncQueryDataSupport` sounds better.

> Also, do you need one for variable for both plugins?

My thinking around having a toggle per plugin was in case one plugin runs into issues with async queries but not the other one, it could be toggled off for just one until it's fixed.

sunker commented:
> Cool, didn't know that. Is this more for documenting it since it's used for AMG, i.e. if we removed these toggles, would it not work in AMG?

Yes it would work! But it would be without proper typings for the new toggles. So actually, it's probably good practice to always define the feature toggles in core grafana (like you've done in this PR) if we're aiming to deploy it to production. I'm also not sure what affect it would have to the feature toggle mechanism in Hosted Grafana if toggles are not defined here. You can read more about feature toggles in Hosted Grafana and how to deploy them [here](https://github.com/grafana/deployment_tools/tree/master/ksonnet/environments/hosted-grafana/waves/feature-toggles).   

> I've updated to a different suffix, AsyncQueryDataSupport sounds better.

Don't think the changes are pushed yet. 

> My thinking around having a toggle per plugin was in case one plugin runs into issues with async queries but not the other one, it could be toggled off for just one until it's fixed.

Okay, makes sense!
